File: src/main/java/berkerdemoglu/engine/graphics/math/geometry/ProjectionSettings.java
package engine.graphics.math.geometry;

public class ProjectionSettings {
	public double width, height;

	public double fov;
	public double near;
	public double far;

	public ProjectionSettings(int width, int height, double fov, double near, double far) {
		this.width = (double) width;
		this.height = (double) height;
		this.fov = fov;
		this.near = near;
		this.far = far;
	}

	@Override
	public ProjectionSettings clone() {
		return new ProjectionSettings((int) width, (int) height, fov, near, far);
	}
}

File: python_api/src/api/geom/projection_settings.py
from ..base import BaseAPIClass
from ..settings import Settings


class ProjectionSettings(BaseAPIClass):
	"""Represents the berkerdemoglu.engine.graphics.math.geometry.ProjectionSettings class."""

	def __init__(self, settings: Settings, fov: float, near: float, far: float):
		self.width = settings.screen_width
		self.height = settings.screen_height
		self.fov = float(fov)
		self.near = float(near)
		self.far = float(far)

	def as_dict(self):
		return self.__dict__
